### PR TITLE
Voucher API endpoint

### PR DESCRIPTION
**Pull request by vkaracic**
Voucher endpoint for displaying all the vouchers. 

Added a `code` filter for easy retrieval by frontend. 
https://openedx.atlassian.net/browse/SOL-1649

### CODE REVIEW

**Comment by vkaracic** [comment]
@mjfrey @marjev please take a look.

**Comment by marjev** [comment]
lgtm :+1:

**Comment by edx-webhook** [comment]
Thanks for the pull request, @vkaracic! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fecommerce&number=503

**Comment by vkaracic** [comment]
@mjfrey @marjev 
I've added the filtering and tests.

**Comment by marjev** [comment]
LGTM

File: ecommerce/extensions/api/v2/tests/views/test_vouchers.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        self.voucher = VoucherFactory(code=self.coupon_code)
+        self.voucher.offers.add(ConditionalOfferFactory(name='test2'))
+
+    def test_voucher_listing(self):
+        """ Verify the endpoint lists out all vouchers. """
+        response = self.client.get(self.path)
+        response_data = json.loads(response.content)
+
+        self.assertEqual(response_data['count'], 2)
+        self.assertEqual(response_data['results'][1]['code'], self.coupon_code)
```
Comment (mjfrey) [inline ecommerce/extensions/api/v2/tests/views/test_vouchers.py:31]: We should compare self and not elements.

File: ecommerce/extensions/api/v2/tests/views/test_vouchers.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    def test_voucher_filtering(self):
+        """ Verify the endpoint filters by code. """
+        filter_path = '{}?code={}'.format(self.path, self.coupon_code)
+        response = self.client.get(filter_path)
+        response_data = json.loads(response.content)
+
+        self.assertEqual(response_data['count'], 1)
+        self.assertEqual(response_data['results'][0]['code'], self.coupon_code)
```
Comment (mjfrey) [inline ecommerce/extensions/api/v2/tests/views/test_vouchers.py:40]: same here.

**Comment by edx-webhook** [comment]
Thanks for the pull request, @vkaracic! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fecommerce&number=503

File: ecommerce/extensions/api/v2/tests/views/test_vouchers.py (hunk 1 above)
Comment (vkaracic) [inline ecommerce/extensions/api/v2/tests/views/test_vouchers.py:31]: You mean changing to:

```
self.assertEqual(response_data['results'][1]['code'], self.voucher.code)
```

?

**Comment by mjfrey** [comment]
:+1:

**Comment by mjfrey** [comment]
Please rebase and squash.

**Pull request merged**